City-of-Helsinki/hel-fi-drupal-grants
# feat: AU-210: Real-time application timeout message

# [AU-210](https://helsinkisolutionoffice.atlassian.net/browse/AU-210)

## What was done
This pull request implements real-time application timeout messages for the application forms.

![image](https://github.com/City-of-Helsinki/hel-fi-drupal-grants/assets/117807015/786e7afe-6fdf-4a24-bac0-5950fceb2dbe)

## How to install
Make sure your instance is up and running on the correct branch.
  * `git checkout feature/AU-210-realtime-timeout-messages`
  * `make fresh`
  * `make drush-cr`

## How to test
1. Login with a test user and start applying for any application.

2. Simultaneously login with an admin user and set the applications closing period to be withing a few minutes of the current time.

3. Observe the application with the test user. A warning message should slide in when the application period has closed.

4. Review the code.


[AU-210]: https://helsinkisolutionoffice.atlassian.net/browse/AU-210?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
